Task2/usersAPI.ts

```diff
@@ -0,0 +1,85 @@
+import express from 'express';
+import uuidv4 from 'uuid/v4';
+
+import { usersCollection } from './usersCollection';
+import validateUser from './checkValidation';
+
+const app: express.Application = express();
+const router = express.Router();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+app.use('/users', router);
+
+function getAutoSuggestUsers (loginSubstringin:string, limit: string):any {
+	if (!loginSubstringin || loginSubstringin.length === 0) return usersCollection;
+	loginSubstringin = loginSubstringin.toLowerCase();
+	const numLimit: number = limit ? +limit : usersCollection.length;
+
+	const filteredAndSortedUsersCollection = usersCollection.filter(user => {
+		return user.login.startsWith(loginSubstringin);
```

Comment:
Here, `user.login`needs to be converted to lowercase too. Because now if you can pass **john** as `loginSubstring` - **John** will be ignored.